### DISCUSSION TITLE
NO_SECRET error on iisnode

### DISCUSSION THREAD

**Discussion by hagenderouen** [nextauthjs/next-auth#3983] [Help]
I'm getting [next-auth][error][NO_SECRET] when deploying to iisnode. I have the NEXTAUTH_SECRET explicitly defined in [...nextauth].js. I run next build and move my .next file to issnode server. I don't get any errors in development. Not sure what's going on here. 

./pages/api/auth/[...nextauth].js
(shortened for brevity)
`export default NextAuth({
  secret: process.env.NEXTAUTH_SECRET,
  providers: [...]
})`

**Comment by sarveshc** [comment]
Dear, I am facing same problem. 

What was the solution?

Please help me out.

You can save my life dear.
